challengeutils/synapseservices/base_service.py

```diff
@@ -172,21 +208,18 @@ def to_dict(self):
 
         return result
 
-    def to_str(self):
-        """Returns the string representation of the model
-
-        :rtype: str
-        """
+    def to_str(self) -> str:
+        """Returns the string representation of the model"""
         return pprint.pformat(self.to_dict())
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         """For `print` and `pprint`"""
         return self.to_str()
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
         """Returns true if both objects are equal"""
         return self.__dict__ == other.__dict__
 
-    def __ne__(self, other):
+    def __ne__(self, other) -> bool:
         """Returns true if both objects are not equal"""
         return not self == other
```

Comment:
Interesting... wouldn't `return self != other` be more readable...?

Reply:
See this post: https://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq.  Seems that I should actually remove this for python3.  This was legacy python2 code.